rflx/expression.py

```diff
@@ -204,7 +208,11 @@ def __init__(self, expr: Expr) -> None:
         self.expr = expr
 
     def __str__(self) -> str:
-        return f"not {self.parenthesized(self.expr)}"
+        try:
+            return self.__str
+        except AttributeError:
+            self.__str = intern(f"not {self.parenthesized(self.expr)}")
+            return self.__str
```

Comment:
What do you think about replacing the try-except block by a decorator (similar to `fatalexceptions` in `parser.grammar`)?

Reply:
Since we're accessing a private member I would have to disable this warning in the decorator. I couldn't find another good way to implement a decorator that can access private members. Also I'd like to test the performance impact.

Reply:
I think disabling the warning would be fine in that case. I agree with you that the performance impact should be tested.

Reply:
Fixed as discussed separately in https://github.com/Componolit/RecordFlux/pull/390/commits/d90437a2dacd0545f541b12fc50169fdcbd6e435.